test/e2e/framework/framework.go

```diff
@@ -117,6 +120,7 @@ func newFramework(kubeconfig string, qps float32, burst int) (*Framework, error)
 const (
 	kubeconfigFlag      = "kubeconfig"
 	gsimageFlag         = "gameserver-image"
+	gsimageTCPUDPFlag   = "gameserver-image-tcpudp"
```

Comment:
This might be a good reason to wrap the functionality into the gameserver-image - wdyt?

Reply:
Yes, I was also not happy with the way this specific change was done.

My reasoning of making a new example was that there already is a udp and a tcp example. I'm on board with making it a flag instead. Will make a commit with this.

Reply:
Yeah, I 100% agree. Just thinking that we already have one image that we use for testing - seems like a lot of work to add another one.

@aLekSer wdyt?

Reply:
OK, I am totally agree with having one example for testing, with a flag passed through to `simple-udp` to enable TCP listener. So we don't need this `gsimageTCPUDPFlag ` in this case, we could turn on the TCP in a specific E2E test.

Reply:
I wonder if we should change the same of our example servers (and consolidate them). Right now this flag defaults to `gcr.io/agones-images/udp-server:0.21` but it's going to be confusing if we add tcp functionality to the _udp_ server. What about renaming it to `simple-game-server` and collapsing https://github.com/googleforgames/agones/tree/master/examples/simple-tcp in with https://github.com/googleforgames/agones/tree/master/examples/simple-udp?

Reply:
Maybe call it something like "simple-server" ?

Works for me :+1:

Reply:
Ah, missing your comment @roberthbailey - "simple-game-server" also works for me as well. Don't mind. :+1:

Reply:
Makes sense, although I'm a bit apprehensive to make the changes as there are likely to be a bunch of places it needs to change, and there could be some hidden ones I would be missing.

Reply:
The simplest might be to publish the image with two names for now until we clean up the other references.

Reply:
Also doesn't have to be done all in this PR as well. We can take multiple steps :+1:

Reply:
For history sake - I've opened this PR to do exactly that: https://github.com/googleforgames/agones/pull/1771